poetry/orb.yaml

```diff
@@ -26,16 +26,24 @@ commands:
         type: string
     steps:
       - run:
-          name: install curl
+          name: Install curl
           command: |
-            if which apk >/dev/null; then
-              apk add --no-cache --no-progress curl
-            elif which apt-get >/dev/null; then
-              apt-get update -qy
-              apt-get install -qy curl
+            if cat /etc/issue | grep "Alpine" >/dev/null 2>&1; then
+              if [ "$ID" = 0 ]; then export SUDO=""; else export SUDO="sudo"; fi
             else
-              echo >&2 "ERROR: could not find supported package manager"
-              exit 1
+              if [[ $EUID == 0 ]]; then export SUDO=""; else export SUDO="sudo"; fi
+            fi
+            echo "export SUDO=$SUDO" >> ${BASH_ENV}
+            if [ ! "$(command -v curl)" ]; then
```

Comment:
Let's move this check to be the first line of this step: there's no need to do _any_ work in this step if `curl` is already installed!

Reply:
Good point